pkg/queue/concurrency_state.go

```diff
@@ -115,11 +130,39 @@ func concurrencyStateRequest(endpoint string, action string) func() error {
 }
 
 // Pause sends a pause request to the concurrency state endpoint.
-func Pause(endpoint string) func() error {
+func Pause(endpoint string) func(*Token) error {
 	return concurrencyStateRequest(endpoint, "pause")
 }
 
 // Resume sends a resume request to the concurrency state endpoint.
-func Resume(endpoint string) func() error {
+func Resume(endpoint string) func(*Token) error {
 	return concurrencyStateRequest(endpoint, "resume")
 }
+
+type Token struct {
```

Comment:
Might be worth using atomic.Value for this: https://pkg.go.dev/go.uber.org/atomic#Value